sqlx-core/src/migrate/error.rs

```diff
@@ -15,9 +15,6 @@ pub enum MigrateError {
     #[error("migration {0} was previously applied but has been modified")]
     VersionMismatch(i64),
 
-    #[error("cannot mix reversible migrations with simple migrations. All migrations should be reversible or simple migrations")]
-    InvalidMixReversibleAndSimple,
```

Comment:
It's actually only just occurred to me, but the removal of this variant is a breaking change. Please restore it and just mark it deprecated, I'll remove it in the next breaking release.